src/ManagedArray.hpp

```diff
@@ -212,6 +212,37 @@ class ManagedArray {
   uint m_elems;
 };
 
+/*!
+ * \brief Construct a ManagedArray from an externally allocated pointer.
+ *
+ * The pointer can exist in any valid ExecutionSpace, and can either be "owned"
+ * or "unowned". An owned pointer will be deallocated by the ArrayManager when
+ * free is called on the returned ManagedArray object.
+ *
+ * \param data Pointer to the raw data.
+ * \param elems Number of elements in the data pointer.
+ * \param space ExecutionSpace where the data pointer exists.
+ * \param owned If true, data will be deallocated by the ArrayManager.
+ *
+ * \tparam T Type of the raw data.
+ *
+ * \return A new ManagedArray containing the raw data pointer.
+ */
+template <typename T>
+ManagedArray<T> 
+makeManagedArray(
```

Comment:
This method could go somewhere else, or even be a static member of the `ManagedArray` class. Thoughts? @rhornung67 @rfhaque